BuildScripts/mvBESFunctions.sql

```diff
@@ -235,6 +236,93 @@ begin
+            (
+                pOldTableName           IN      TEXT,
+                pNewTableName           IN      TEXT,
+                pOwner                  IN      TEXT        DEFAULT USER
+            )
+    RETURNS VOID
+AS
+$BODY$
+/* ---------------------------------------------------------------------------------------------------------------------------------
+Routine Name: mv$renameMaterializedViewLog
+Author:       Ethan McQuarrie
+Date:         06/12/2021
+------------------------------------------------------------------------------------------------------------------------------------
+Revision History    Push Down List
+------------------------------------------------------------------------------------------------------------------------------------
+Date        | Name          | Description
+------------+---------------+-------------------------------------------------------------------------------------------------------
+            |               |
+------------+---------------+-------------------------------------------------------------------------------------------------------
+Description:    Renames a materialized view log table
+
+Arguments:      IN      pOldTableName       The name of the base table the materialized view logs are against
+                IN      pNewTableName       The name this table is being updated to
+                IN      pOwner              Optional, the owner of the materialized view, defaults to user
+Returns:                VOID
+
+************************************************************************************************************************************
+Copyright 2021 Beyond Essential Systems Pty Ltd
+***********************************************************************************************************************************/
+DECLARE
+
+    rConst          mv$allConstants;
+    aViewLog        pg$mview_logs;
+
+    tRenameLogTableSqlStatement             TEXT;
+    tUpdateMViewLogsTableSqlStatement       TEXT;
+    tUpdateMViewTableColumns                TEXT;
+BEGIN
+
+    rConst   := mv$buildAllConstants();
+    tRenameLogTableSqlStatement         := rConst.ALTER_TABLE || 'log$_' || pOldTableName || rConst.RENAME_TO_COMMAND || 'log$_' || pNewTableName;
+    tUpdateMViewLogsTableSqlStatement   := rConst.UPDATE_COMMAND || 'pg$mview_logs' || rConst.SET_COMMAND
+                                            || 'table_name = ' || rConst.SINGLE_QUOTE_CHARACTER || pNewTableName || rConst.SINGLE_QUOTE_CHARACTER || rConst.COMMA_CHARACTER
+                                            || 'pglog$_name = ' || rConst.SINGLE_QUOTE_CHARACTER || 'log$_' || pNewTableName || rConst.SINGLE_QUOTE_CHARACTER
+                                            || rConst.WHERE_COMMAND || 'table_name = ' || rConst.SINGLE_QUOTE_CHARACTER || pOldTableName || rConst.SINGLE_QUOTE_CHARACTER;
+    tUpdateMViewTableColumns            := rConst.UPDATE_COMMAND || 'pg$mviews' || rConst.SET_COMMAND
```

Comment:
In this update I'm a bit confused - shouldn't we also need to update `select_columns` and `table_names`?

Reply:
Yeah that's kind of an weird case. Those fields are filled out from the `build_analytics_table` function, which we have more of a precedent of updating directly when we need to. In the [migration where I use this](https://github.com/beyondessential/tupaia/pull/3558/files#diff-737366c605d0200d2db03b9c986d64df1d9363ef90a1bb60f4eb841f1624a110R192) I update that function first, then run this to update the other references to the table name. I'd argue that it's safer that way too, since those columns aren't _just_ simple table/column names like the others, running a string replace on them has more chance of accidentally breaking something

Reply:
Cool that all makes sense, and yes does sound safer. It makes me wonder about adding this util to the `pg-mv-fast-refresh` package though, as that means it isn't standalone and could be misused in future. Perhaps we should just run the content once as part of the patch instead? Or if we want the util in here, adding some qualifying comments at the top seems a good idea.

Reply:
I've added a comment to the code explaining this fact